### ISSUE TITLE
LayoutView will unbind all DOM events when deactivating a retained view

### ISSUE THREAD

**Issue by jasonwebster** [walmartlabs/thorax#332]
I'm using a LayoutView where I'm retaining the views, so they are not destroyed on change. However, `setView` will still nuke all DOM events bound to the view's element due to it calling `oldView.$el.remove()`.

2 lines later, it also calls `this._removeChild(oldView)` which _will_ call `_destroy` on the old view if the retain count is zero (which in turn calls `Backbone.View.prototype.remove()`, which in turn calls `$el.remove`).

Current functionality is preserved and reusing view instances is not buggy if we simply change:

`oldView.$el && oldView.$el.remove()`

to

`oldView.$el && oldView.$el.detach()`

https://github.com/walmartlabs/thorax/blob/master/src/layout.js#L33
